doc/design/design_high-available_sqlflow_server.md

```diff
@@ -0,0 +1,232 @@
+# High-available SQLFlow server
+
+## Motivations
+
+In the current system, SQLFlow client connects the SQLFlow server with a long live connection. the SQLFlow client sends
+a gRPC request which contains a SQL statement and blocking until the SQLFlow server finishes executing the SQL statement.
+
+For each SQL statement, the SQLFlow code generator would generate a submitter program in Python, and then the SQLFlow server
+would launch a process on the host or launch a distributed job on a cluster(Kubernetes/Yarn), this would cause two problems in the production environment:
```

Comment:
When the SQLFlow server receives one training SQL statement, it will generate a python training program that runs on the host or submit the training job to some distributed training service cluster (Kubernetes/Yarn). This will cause: